CodeHub.Service/Services/BranchRepositoryService.cs

```diff
@@ -0,0 +1,93 @@
+﻿using AutoMapper;
+using CodeHub.DataAccess.Repositories;
+using CodeHub.Domain.Entities;
+using CodeHub.Model.BranchRepositories;
+using CodeHub.Service.Exceptions;
+using CodeHub.Service.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace CodeHub.Service.Services;
+
+public class BranchRepositoryService : IBranchRepositoryService
+{
+    private IMapper mapper;
+    private IGenericRepository<BranchRepository> Repository;
+    private IUserService userService;
+    private IRepositoryService repositoryService;
+
+
+
+    public BranchRepositoryService(IMapper mapper, IGenericRepository<BranchRepository> Repository, IUserService userService, IRepositoryService repositoryService)
+    {
+        this.mapper = mapper;
+        this.Repository = Repository;
+        this.userService = userService;
+        this.repositoryService = repositoryService;
+    }
+
+    public async Task<BranchRepositoryViewModel> CreateAsync(BranchRepositoryCreateModel branchRepository)
+    {
+        var existUser = await userService.GetByIdAsync(branchRepository.UserId);
+        var existRepository = await repositoryService.GetByIdAsync(branchRepository.UserId);
```

Comment:
You should use branchRepository.RepositoryId

Reply:
+

Reply:
ok